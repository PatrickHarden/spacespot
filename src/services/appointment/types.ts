export interface AppointmentState {
  data?: Array<Appointment>
  error?: string
}

export interface Appointment {
  id: number
  enquireId: number
  description: string
  peer1Id: string
  peer2Id: string
  proposedTime: string
  createdAt: string
  lastUpdatedAt: string
  lastProposedBy: string
  appointmentStatus: string
}

export interface NewAppointment {
  description: string
  proposedTime: string
}
export interface EditAppointment {
  id: string
  description: string
  proposedTime: string
}
export interface AppointmentAction {
  type: string
  payload?: Array<Appointment> | NewAppointment | EditAppointment | string
}
