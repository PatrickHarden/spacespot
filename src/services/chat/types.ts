export interface ChatAction {
  type: string
  payload?: ChatResponse | ChatMessage | string
}
export interface MessageAction {
  type: string
  payload?: string
}
export interface ChatState {
  data?: ChatResponse
  error?: string
}

export interface ChatResponse {
  chatId: number
  enquiryId: number
  peer1Id: string
  peer2Id: string
  messages: Array<ChatMessage>
}

export interface ChatMessage {
  userId: string
  message: string
  sentAt: string
}
