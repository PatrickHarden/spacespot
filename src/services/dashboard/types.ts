import { Space } from 'services/space/types'

export interface DashboardState {
  buildings: Array<Space>
  spaces: { [key: string]: Array<Space> }
  error?: string
  isLoading: boolean
  initOK: boolean
  pendingDelete: Array<string>
}
