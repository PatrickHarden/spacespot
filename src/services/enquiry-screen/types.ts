export interface EnquiryScreenAction {
  type: string
  payload: EnquiryScreenStateData | string
}
export interface EnquiryScreenState {
  data?: EnquiryScreenStateData
  error?: string
}
export interface EnquiryScreenStateData {
  id: string
}
