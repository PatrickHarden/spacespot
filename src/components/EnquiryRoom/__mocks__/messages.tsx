export const messages = [
  {
    userId: 'f9251bff-4527-4ebd-9028-f77b4bd0e0ce',
    message: 'message',
    sentAt: '2018-10-18T12:40:35.886+0000',
  },
  {
    userId: 'f9251bff-4527-4ebd-9028-f77b4bd0e0ce',
    message: 'message1',
    sentAt: '2018-10-18T12:40:43.547+0000',
  },
  {
    userId: 'f9251bff-4527-4ebd-9028-f77b4bd0e0ce',
    message: 'message2',
    sentAt: '2018-10-18T12:40:44.946+0000',
  },
  {
    userId: 'f9251bff-4527-4ebd-9028-f77b4bd0e0ce',
    message: 'message3',
    sentAt: '2018-10-18T12:40:45.678+0000',
  },
  {
    userId: 'f9251bff-4527-4ebd-9028-f77b4bd0e0ce',
    message: 'message4',
    sentAt: '2018-10-18T12:40:46.856+0000',
  },
  {
    userId: 'f9251bff-4527-4ebd-9028-f77b4bd0e0ce',
    message: 'message5',
    sentAt: '2018-10-23T14:17:59.662+0000',
  },
  {
    userId: 'f9251bff-4527-4ebd-9028-f77b4bd0e0ce',
    message: 'jk',
    sentAt: '2018-10-28T10:13:59.805+0000',
  },
  {
    userId: 'f9251bff-4527-4ebd-9028-f77b4bd0e0ce',
    message: 'message6',
    sentAt: '2018-10-28T10:16:16.585+0000',
  },
  {
    userId: 'f9251bff-4527-4ebd-9028-f77b4bd0e0ce',
    message: '11:44',
    sentAt: '2018-10-28T10:45:02.592+0000',
  },
  {
    userId: 'f9251bff-4527-4ebd-9028-f77b4bd0e0ce',
    message: '12:34',
    sentAt: '2018-10-28T11:23:27.467+0000',
  },
  {
    userId: 'f9251bff-4527-4ebd-9028-f77b4bd0e0ce',
    message: 'll',
    sentAt: '2018-10-28T11:29:37.057+0000',
  },
  {
    userId: 'f9251bff-4527-4ebd-9028-f77b4bd0e0ce',
    message: '12:30',
    sentAt: '2018-10-28T11:30:13.643+0000',
  },
  {
    userId: 'f9251bff-4527-4ebd-9028-f77b4bd0e0ce',
    message: 'The message',
    sentAt: '2018-10-29T08:51:21.007+0000',
  },
  {
    userId: 'f9251bff-4527-4ebd-9028-f77b4bd0e0ce',
    message: 'Another',
    sentAt: '2018-10-29T09:28:49.589+0000',
  },
  {
    userId: 'f9251bff-4527-4ebd-9028-f77b4bd0e0ce',
    message: 'Other',
    sentAt: '2018-10-29T09:29:02.900+0000',
  },
]
