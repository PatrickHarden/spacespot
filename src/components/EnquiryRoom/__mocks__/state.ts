export const mockState: any = {
  enquiry: {
    data: {
      spaceId: 'SS_SPACE_665',
      peer1Id: '1',
      peer2Id: '2',
      chatId: 159,
      enquireStatus: 'ACCEPTED',
      negotiationId: 139,
      enquireId: 159,
      flex: false,
    },
  },
  negotiation: {
    data: {
      flexible: false,
      terms: [
        {
          termId: 5,
          value: 'MONTHLY',
        },
        {
          termId: 6,
          value: '2000',
        },
        {
          termId: 7,
          value: '2',
        },
        {
          termId: 8,
          value: '2000',
        },
        {
          termId: 9,
          value: 'GUARANTEE',
        },
        {
          termId: 1,
          value: '3500',
        },
        {
          termId: 2,
          value: '40000',
        },
        {
          termId: 3,
          value: '2020-04-22',
        },
        {
          termId: 4,
          value: '13',
        },
      ],
      negotiationId: 139,
      enquireId: 159,
      createdAt: '2020-02-14T06:30:08.354+0000',
      lastUpdated: '2020-03-05T16:35:02.513+0000',
      lastModifyBy: '0062fb00-e389-4bb8-9119-613e3c45938a',
      status: 'PROPOSED',
      customFitout: false,
      fitoutOption: 0,
      specialProvision: 'gfd',
      specialProvisionStatus: 'CREATED',
    },
    loading: false,
    termsFields: {
      '1': 'Area',
      '2': 'Rent',
      '3': 'Start',
      '4': 'Duration',
      '5': 'Frequency',
      '6': 'AdditionalCosts',
      '7': 'Deposit',
      '8': 'ServiceCharges',
      '9': 'DepositType',
    },
    signerInfo: {
      id: 1,
      name: 'Name1',
      emailId: 'name1@email.com',
      companyName: 'companyA',
      companyNumber: '12345678',
    },
    customFitout: [],
  },
  user: {
    auth: {
      idToken: {
        claims: {
          // eslint-disable-next-line @typescript-eslint/camelcase
          extension_Landlord: true,
        },
      },
    },
  },
}
