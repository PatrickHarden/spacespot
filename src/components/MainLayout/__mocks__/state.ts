export const mockState: any = {
  user: {
    auth: {
      idToken: {
        claims: {
          // eslint-disable-next-line @typescript-eslint/camelcase
          extension_Landlord: true,
        },
      },
    },
  },
}
