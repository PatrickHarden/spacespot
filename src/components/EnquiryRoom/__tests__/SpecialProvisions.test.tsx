import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import { createMockStore } from 'redux-test-utils'
import { Provider } from 'react-redux'

import SpecialProvisions from '../SpecialProvisions'

import { mockState } from '../__mocks__/state'
import { SpecialProvisionStatus } from 'services/negotiation/types'

jest.mock('react-quill', () => {
  const ComponentToMock = (props: any) => <textarea {...props} />
  return ComponentToMock
})

jest.mock('react-intl', () => ({
  useIntl: () => ({ formatMessage: (s: any) => s.id }),
  createIntlCache: () => jest.fn(),
  createIntl: () => ({ formatMessage: (s: any) => s.id }),
}))

const mockDispatch = jest.fn()
const mockNeg = mockState.negotiation.data

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useDispatch: () => mockDispatch,
}))

it('Landlord Empty', () => {
  mockNeg.specialProvisionStatus = SpecialProvisionStatus.Accepted
  mockNeg.specialProvision = ''
  const store: any = createMockStore(mockState)
  const wrapper = render(
    <Provider store={store}>
      <SpecialProvisions disabled={false} />
    </Provider>,
  )
  expect(wrapper.asFragment()).toMatchSnapshot()
  fireEvent.click(wrapper.getByText('SPECIAL_PROV_ADD'))
  fireEvent.change(wrapper.getByLabelText('SPECIAL_PROV_FORM_LABEL'), {
    target: { value: 'Prov1' },
  })
  fireEvent.click(wrapper.getByText('SPECIAL_PROV_CANCEL'))
})

it('Landlord Accepted with content', () => {
  mockNeg.specialProvisionStatus = SpecialProvisionStatus.Accepted
  mockNeg.specialProvision = 'Provision 1'
  const store: any = createMockStore(mockState)
  const wrapper = render(
    <Provider store={store}>
      <SpecialProvisions disabled={false} />
    </Provider>,
  )
  expect(wrapper.asFragment()).toMatchSnapshot()
  fireEvent.click(wrapper.getByText('SPECIAL_PROV_MODIFY'))
  fireEvent.click(wrapper.getByText('SPECIAL_PROV_CANCEL'))
  fireEvent.click(wrapper.getByText('SPECIAL_PROV_MODIFY'))
  fireEvent.click(wrapper.getByText('SPECIAL_PROV_SAVE'))
  fireEvent.click(wrapper.getByTestId('removeSP'))
})

it('Landlord Created', () => {
  mockNeg.specialProvisionStatus = SpecialProvisionStatus.Created
  const store: any = createMockStore(mockState)
  const wrapper = render(
    <Provider store={store}>
      <SpecialProvisions disabled={false} />
    </Provider>,
  )
  expect(wrapper.asFragment()).toMatchSnapshot()
})

it('Landlord Declined', () => {
  mockNeg.specialProvisionStatus = SpecialProvisionStatus.Declined
  const store: any = createMockStore(mockState)
  const wrapper = render(
    <Provider store={store}>
      <SpecialProvisions disabled={false} />
    </Provider>,
  )
  expect(wrapper.asFragment()).toMatchSnapshot()
})

it('Landlord closed', () => {
  mockNeg.specialProvisionStatus = SpecialProvisionStatus.Closed
  const store: any = createMockStore(mockState)
  const wrapper = render(
    <Provider store={store}>
      <SpecialProvisions disabled={false} />
    </Provider>,
  )
  expect(wrapper.asFragment()).toMatchSnapshot()
})

it('Tenant Accepted', () => {
  mockNeg.specialProvisionStatus = SpecialProvisionStatus.Accepted
  mockState.user.auth.idToken.claims['extension_Landlord'] = false
  const store: any = createMockStore(mockState)
  const wrapper = render(
    <Provider store={store}>
      <SpecialProvisions disabled={false} />
    </Provider>,
  )
  expect(wrapper.asFragment()).toMatchSnapshot()
  fireEvent.click(wrapper.getByText('NEGOTIATE_DECLINE'))
})

it('Tenant Declined', () => {
  mockNeg.specialProvisionStatus = SpecialProvisionStatus.Declined
  mockState.user.auth.idToken.claims['extension_Landlord'] = false
  const store: any = createMockStore(mockState)
  const wrapper = render(
    <Provider store={store}>
      <SpecialProvisions disabled={false} />
    </Provider>,
  )
  expect(wrapper.asFragment()).toMatchSnapshot()
  fireEvent.click(wrapper.getByText('NEGOTIATE_ACCEPT'))
})

it('Tenant Empty', () => {
  mockNeg.specialProvisionStatus = SpecialProvisionStatus.Accepted
  mockState.user.auth.idToken.claims['extension_Landlord'] = false
  mockNeg.specialProvision = ''
  const store: any = createMockStore(mockState)
  const wrapper = render(
    <Provider store={store}>
      <SpecialProvisions disabled={false} />
    </Provider>,
  )
  expect(wrapper.asFragment()).toMatchSnapshot()
})

it('Tenant created', () => {
  mockNeg.specialProvisionStatus = SpecialProvisionStatus.Created
  mockState.user.auth.idToken.claims['extension_Landlord'] = false
  mockNeg.specialProvision = 'Test'
  const store: any = createMockStore(mockState)
  const wrapper = render(
    <Provider store={store}>
      <SpecialProvisions disabled={false} />
    </Provider>,
  )
  expect(wrapper.asFragment()).toMatchSnapshot()
})

it('Tenant disabled', () => {
  mockNeg.specialProvisionStatus = SpecialProvisionStatus.Accepted
  mockState.user.auth.idToken.claims['extension_Landlord'] = false
  mockNeg.specialProvision = 'Test'
  const store: any = createMockStore(mockState)
  const wrapper = render(
    <Provider store={store}>
      <SpecialProvisions disabled={true} />
    </Provider>,
  )
  expect(wrapper.asFragment()).toMatchSnapshot()
})
