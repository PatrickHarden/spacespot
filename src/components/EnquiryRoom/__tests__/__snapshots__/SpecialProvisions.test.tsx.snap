// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landlord Accepted with content 1`] = `
<DocumentFragment>
  <div
    class="sc-cSHVUG jkVRhX"
  >
    <div
      class="sc-gzVnrw gGBuAk"
    >
      <p>
        <span>
          SPECIAL_PROV_TEMPLATE_MSG: 
        </span>
        <button
          class="sc-dnqmqq jAHHlC"
          data-testid="download"
        >
          SPECIAL_PROV_TEMPLATE_STANDARD
        </button>
      </p>
      <div>
        <div
          class=""
          tabindex="0"
        >
          <input
            accept="application/pdf"
            autocomplete="off"
            multiple=""
            style="display: none;"
            tabindex="-1"
            type="file"
          />
          <button
            class="sc-htoDjs kHlInK"
            data-testid="upload"
          >
            <img
              alt="upload"
              src="upload.svg"
            />
            SPECIAL_PROV_TEMPLATE_UPLOAD
          </button>
        </div>
      </div>
    </div>
    <div
      class="sc-chPdSV idJAuf"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root sc-kgoBCf keUeIv"
        data-testid="removeSP"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
        />
      </svg>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root sc-bZQynM jgQicF"
        color="#6CB9D5"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
        />
      </svg>
      <div>
        Provision 1
      </div>
    </div>
    <div
      class="sc-ckVGcZ grMCQK"
    >
      <button
        class="sc-bdVaJa sc-iwsKbI fbhcGV"
      >
        SPECIAL_PROV_MODIFY
      </button>
      <span
        class="sc-dxgOiQ dGLNsP"
      >
        SPECIAL_PROV_STATUS_ACCEPTED
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Landlord Created 1`] = `
<DocumentFragment>
  <div
    class="sc-cSHVUG jkVRhX"
  >
    <div
      class="sc-gzVnrw gGBuAk"
    >
      <p>
        <span>
          SPECIAL_PROV_TEMPLATE_MSG: 
        </span>
        <button
          class="sc-dnqmqq jAHHlC"
          data-testid="download"
        >
          SPECIAL_PROV_TEMPLATE_STANDARD
        </button>
      </p>
      <div>
        <div
          class=""
          tabindex="0"
        >
          <input
            accept="application/pdf"
            autocomplete="off"
            multiple=""
            style="display: none;"
            tabindex="-1"
            type="file"
          />
          <button
            class="sc-htoDjs kHlInK"
            data-testid="upload"
          >
            <img
              alt="upload"
              src="upload.svg"
            />
            SPECIAL_PROV_TEMPLATE_UPLOAD
          </button>
        </div>
      </div>
    </div>
    <div
      class="sc-chPdSV kQTYaE"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root sc-kgoBCf keUeIv"
        data-testid="removeSP"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
        />
      </svg>
      <div>
        Provision 1
      </div>
    </div>
    <div
      class="sc-ckVGcZ grMCQK"
    >
      <button
        class="sc-bdVaJa sc-iwsKbI fbhcGV"
      >
        SPECIAL_PROV_MODIFY
      </button>
      <span
        class="sc-dxgOiQ cnsbFl"
      >
        SPECIAL_PROV_STATUS_CREATED
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Landlord Declined 1`] = `
<DocumentFragment>
  <div
    class="sc-cSHVUG jkVRhX"
  >
    <div
      class="sc-gzVnrw gGBuAk"
    >
      <p>
        <span>
          SPECIAL_PROV_TEMPLATE_MSG: 
        </span>
        <button
          class="sc-dnqmqq jAHHlC"
          data-testid="download"
        >
          SPECIAL_PROV_TEMPLATE_STANDARD
        </button>
      </p>
      <div>
        <div
          class=""
          tabindex="0"
        >
          <input
            accept="application/pdf"
            autocomplete="off"
            multiple=""
            style="display: none;"
            tabindex="-1"
            type="file"
          />
          <button
            class="sc-htoDjs kHlInK"
            data-testid="upload"
          >
            <img
              alt="upload"
              src="upload.svg"
            />
            SPECIAL_PROV_TEMPLATE_UPLOAD
          </button>
        </div>
      </div>
    </div>
    <div
      class="sc-kGXeez dtpVJJ"
    >
      SPECIAL_PROV_REJECT_MSG
    </div>
    <div
      class="sc-chPdSV dGEpEe"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root sc-kgoBCf keUeIv"
        data-testid="removeSP"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
        />
      </svg>
      <div>
        Provision 1
      </div>
    </div>
    <div
      class="sc-ckVGcZ grMCQK"
    >
      <button
        class="sc-bdVaJa sc-iwsKbI fbhcGV"
      >
        SPECIAL_PROV_MODIFY
      </button>
      <span
        class="sc-dxgOiQ bhcTgB"
      >
        SPECIAL_PROV_STATUS_REJECTED
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Landlord Empty 1`] = `
<DocumentFragment>
  <div
    class="sc-cSHVUG jkVRhX"
  >
    <div
      class="sc-gzVnrw gGBuAk"
    >
      <p>
        <span>
          SPECIAL_PROV_TEMPLATE_MSG: 
        </span>
        <button
          class="sc-dnqmqq jAHHlC"
          data-testid="download"
        >
          SPECIAL_PROV_TEMPLATE_STANDARD
        </button>
      </p>
      <div>
        <div
          class=""
          tabindex="0"
        >
          <input
            accept="application/pdf"
            autocomplete="off"
            multiple=""
            style="display: none;"
            tabindex="-1"
            type="file"
          />
          <button
            class="sc-htoDjs kHlInK"
            data-testid="upload"
          >
            <img
              alt="upload"
              src="upload.svg"
            />
            SPECIAL_PROV_TEMPLATE_UPLOAD
          </button>
        </div>
      </div>
    </div>
    <h3>
      SPECIAL_PROV_HEADING
    </h3>
    <ul>
      <li>
        SPECIAL_PROV_INDEX
      </li>
      <li>
        SPECIAL_PROV_DOCS
      </li>
    </ul>
    <div
      class="sc-ckVGcZ grMCQK"
    >
      <button
        class="sc-bdVaJa sc-iwsKbI fbhcGV"
      >
        <svg
          height="1.1em"
          viewBox="0 0 18 17"
          width="1.1em"
        >
          <g
            fill="none"
            fill-rule="evenodd"
            stroke="#6CB9D5"
            stroke-linecap="round"
            stroke-width="2"
          >
            <path
              d="M9 1.6v13.8M1.64 8.5h14.72"
            />
          </g>
        </svg>
        SPECIAL_PROV_ADD
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Landlord closed 1`] = `
<DocumentFragment>
  <div
    class="sc-cSHVUG jkVRhX"
  >
    <div
      class="sc-gzVnrw gGBuAk"
    >
      <p>
        <span>
          SPECIAL_PROV_TEMPLATE_MSG: 
        </span>
        <button
          class="sc-dnqmqq jAHHlC"
          data-testid="download"
        >
          SPECIAL_PROV_TEMPLATE_STANDARD
        </button>
      </p>
      <div>
        <div
          class=""
          tabindex="0"
        >
          <input
            accept="application/pdf"
            autocomplete="off"
            multiple=""
            style="display: none;"
            tabindex="-1"
            type="file"
          />
          <button
            class="sc-htoDjs kHlInK"
            data-testid="upload"
          >
            <img
              alt="upload"
              src="upload.svg"
            />
            SPECIAL_PROV_TEMPLATE_UPLOAD
          </button>
        </div>
      </div>
    </div>
    <div
      class="sc-chPdSV bBbhVf"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root sc-kgoBCf keUeIv"
        data-testid="removeSP"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
        />
      </svg>
      <div>
        Provision 1
      </div>
    </div>
    <div
      class="sc-ckVGcZ grMCQK"
    >
      <button
        class="sc-bdVaJa sc-iwsKbI fbhcGV"
      >
        SPECIAL_PROV_MODIFY
      </button>
      <span
        class="sc-dxgOiQ dTQwIU"
      >
        SPECIAL_PROV_STATUS_CREATED
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Tenant Accepted 1`] = `
<DocumentFragment>
  <div
    class="sc-cSHVUG jkVRhX"
  >
    <div
      class="sc-gzVnrw gGBuAk"
    >
      <p>
        <span>
          SPECIAL_PROV_TEMPLATE_MSG: 
        </span>
        <button
          class="sc-dnqmqq jAHHlC"
          data-testid="download"
        >
          SPECIAL_PROV_TEMPLATE_STANDARD
        </button>
      </p>
    </div>
    <div
      class="sc-chPdSV idJAuf"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root sc-kgoBCf dYkZbJ"
        data-testid="removeSP"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
        />
      </svg>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root sc-bZQynM jgQicF"
        color="#6CB9D5"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
        />
      </svg>
      <div>
        Provision 1
      </div>
    </div>
    <div
      class="sc-jKJlTe glIfdv"
    >
      <div>
        <button
          class="sc-fjdhpX jwFMl"
        >
          NEGOTIATE_DECLINE
        </button>
      </div>
      <div>
        <div>
          <span
            class="sc-dxgOiQ dGLNsP"
          >
            SPECIAL_PROV_STATUS_ACCEPTED
          </span>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Tenant Declined 1`] = `
<DocumentFragment>
  <div
    class="sc-cSHVUG jkVRhX"
  >
    <div
      class="sc-gzVnrw gGBuAk"
    >
      <p>
        <span>
          SPECIAL_PROV_TEMPLATE_MSG: 
        </span>
        <button
          class="sc-dnqmqq jAHHlC"
          data-testid="download"
        >
          SPECIAL_PROV_TEMPLATE_STANDARD
        </button>
      </p>
    </div>
    <div
      class="sc-kGXeez dtpVJJ"
    >
      SPECIAL_PROV_TENANT_REJECT_MSG
    </div>
    <div
      class="sc-chPdSV dGEpEe"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root sc-kgoBCf dYkZbJ"
        data-testid="removeSP"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
        />
      </svg>
      <div>
        Provision 1
      </div>
    </div>
    <div
      class="sc-jKJlTe glIfdv"
    >
      <div>
        <button
          class="sc-fjdhpX jwFMl"
        >
          NEGOTIATE_ACCEPT
        </button>
      </div>
      <div>
        <div>
          <span
            class="sc-dxgOiQ bhcTgB"
          >
            SPECIAL_PROV_STATUS_REJECTED
          </span>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Tenant Empty 1`] = `
<DocumentFragment>
  <div
    class="sc-cSHVUG jkVRhX"
  >
    <div
      class="sc-gzVnrw gGBuAk"
    >
      <p>
        <span>
          SPECIAL_PROV_TEMPLATE_MSG: 
        </span>
        <button
          class="sc-dnqmqq jAHHlC"
          data-testid="download"
        >
          SPECIAL_PROV_TEMPLATE_STANDARD
        </button>
      </p>
    </div>
    <div
      class="sc-jKJlTe glIfdv"
    >
      <div />
      <div>
        <div>
          <span
            class="sc-dxgOiQ dTQwIU"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Tenant created 1`] = `
<DocumentFragment>
  <div
    class="sc-cSHVUG jkVRhX"
  >
    <div
      class="sc-gzVnrw gGBuAk"
    >
      <p>
        <span>
          SPECIAL_PROV_TEMPLATE_MSG: 
        </span>
        <button
          class="sc-dnqmqq jAHHlC"
          data-testid="download"
        >
          SPECIAL_PROV_TEMPLATE_STANDARD
        </button>
      </p>
    </div>
    <div
      class="sc-chPdSV kQTYaE"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root sc-kgoBCf dYkZbJ"
        data-testid="removeSP"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
        />
      </svg>
      <div>
        Test
      </div>
    </div>
    <div
      class="sc-jKJlTe glIfdv"
    >
      <div />
      <div>
        <button
          class="sc-bdVaJa sc-VigVT epVYbe"
        >
          NEGOTIATE_DECLINE
        </button>
        <button
          class="sc-bwzfXH sc-jTzLTM gaZteH"
        >
          NEGOTIATE_ACCEPT
        </button>
      </div>
    </div>
    <div
      class="sc-kpOJdX eVdXhQ"
    >
      NEGOTIATE_SPECIAL_PROVISION_ACCEPTING
    </div>
  </div>
</DocumentFragment>
`;

exports[`Tenant disabled 1`] = `
<DocumentFragment>
  <div
    class="sc-cSHVUG jkVRhX"
  >
    <div
      class="sc-gzVnrw gGBuAk"
    >
      <p>
        <span>
          SPECIAL_PROV_TEMPLATE_MSG: 
        </span>
        <button
          class="sc-dnqmqq jAHHlC"
          data-testid="download"
        >
          SPECIAL_PROV_TEMPLATE_STANDARD
        </button>
      </p>
    </div>
    <div
      class="sc-chPdSV idJAuf"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root sc-kgoBCf dYkZbJ"
        data-testid="removeSP"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
        />
      </svg>
      <div>
        Test
      </div>
    </div>
  </div>
</DocumentFragment>
`;
